src/ripple/app/ledger/MetaView.h

```diff
@@ -150,41 +238,79 @@ class MetaView : public View
     void
     update (std::shared_ptr<SLE> const& sle) override;
 
-    bool
-    openLedger() const override;
-
     //--------------------------------------------------------------------------
 
-    /** Apply changes to the parent View */
+    /** Apply changes to the base View.
+
+        `to` must contain contents identical to the
+        parent view passed upon construction, else
+        undefined behavior will result.
+
+        After a call to apply(), the only valid operation
+        on the object is a call to the destructor.
+    */
     void
-    apply();
+    apply (BasicView& to,
+        beast::Journal j = {});
+
```

Comment:
Crazy idea: An object after a call to `apply` is similar to a moved-from object. What if we declared apply `apply ()&&` and called it `std::move(o).apply()`?